src/huggingface_hub/utils/cache_explorer.py

```diff
@@ -0,0 +1,246 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Literal, Optional, Set, Union
+
+from ..constants import HUGGINGFACE_HUB_CACHE
+
+
+@dataclass
+class CachedFileInfo:
+    """Information about a single file.
+
+    The file path is a symlink existing in the `snapshots` folder and referring to a
+    blob in the `blobs` folder.
+    """
+
+    file_name: str
+    file_path: Path
+    size_on_disk: int
+    blob_path: Path
+
+    @property
+    def size_on_disk_str(self) -> str:
+        return _format_size(self.size_on_disk)
+
+
+@dataclass
+class CachedRevisionInfo:
+    r"""Information about a revision (a snapshot).
+
+    A revision can be either referenced by 1 or more `refs` or be "detached" (no refs).
+
+    /!\ `size_on_disk` is not necesarily the sum of all file sizes because of possible
+        duplicated files. Also only blobs are taken into account, not the (neglectable)
+        size of folders and symlinks.
```

Comment:
```suggestion
    <Tip warning={true}>

    `size_on_disk` is not necesarily the sum of all file sizes because of possible
    duplicated files. Also only blobs are taken into account, not the (neglectable)
    size of folders and symlinks.

    </Tip>
```
If you want this to be rendered as a warning block in the doc.

Reply:
Thanks for the review and the tips ! 👍 

I'll make the changes when I come back to this PR. I first wait for other feedback and then will properly write the CLI utils + tests + proper docstring + the remaining stuff.